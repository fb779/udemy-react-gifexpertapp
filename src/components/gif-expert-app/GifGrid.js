import React, {useEffect, useState} from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';
import {getGif} from '../../helpers/ApiGetGifs';

import './gifgridapp.css';

const GifGrid = ({category}) => {
  const [images, setImages] = useState([]);

  // useEffect(async () => {
  //   setImages(await getGif(category));
  // }, []);
  useEffect(() => {
    getGif(category).then(setImages);
  }, [category]);

  return (
    <section className="gif">
      <section className="gif-title">
        <h4>{category}</h4>
      </section>

      <section className="gif-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </section>
    </section>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
