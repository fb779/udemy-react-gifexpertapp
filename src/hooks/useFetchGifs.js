import {useEffect, useState} from 'react';
import {getGifs} from '../helpers/ApiGetGifs';

export const useFetchGifs = (category, delay = 0) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((data) =>
      setTimeout(() => {
        setState({data, loading: false});
      }, delay)
    );
  }, [category, delay]);

  return state;
};
