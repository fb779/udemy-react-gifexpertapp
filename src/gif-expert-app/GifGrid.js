import React from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';

import {useFetchGifs} from '../hooks/useFetchGifs';

import './gifgridapp.css';

const GifGrid = ({category}) => {
  const {data: images, loading} = useFetchGifs(category);

  return (
    <section className="gif">
      <section className="gif-title animate__animated animate__lightSpeedInRight">
        <h4>{category}</h4>
      </section>

      {loading && <p className="animate__animated animate__flash">Cargando data...</p>}

      <section className="gif-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </section>
    </section>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
