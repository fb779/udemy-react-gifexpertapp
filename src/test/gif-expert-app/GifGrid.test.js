import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../gif-expert-app/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
  const category = 'Samurai X';
  // let wrapper = shallow(<GifGrid category={category} />);
  // let wrapper;

  // beforeEach(() => {
  //   useFetchGifs.mockReturnValue({
  //     data: [],
  //     loading: true,
  //   });
  //   wrapper = shallow(<GifGrid category={category} />);
  // });

  test('Prueba de carga del componente ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs ', () => {
    const gifs = [
      {
        id: 'ABCD',
        title: 'cualquier cosa',
        url: 'http://localhost/cualquier/cosa.png',
      },
      {
        id: 'edfr',
        title: 'cualquier cosa',
        url: 'http://localhost/cualquier/cosa.png',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
