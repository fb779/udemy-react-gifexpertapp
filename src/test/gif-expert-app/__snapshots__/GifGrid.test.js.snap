// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan imagenes con useFetchGifs  1`] = `
<section
  className="gif"
>
  <section
    className="gif-title animate__animated animate__lightSpeedInRight"
  >
    <h4>
      Samurai X
    </h4>
  </section>
  <section
    className="gif-grid"
  >
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="cualquier cosa"
      url="http://localhost/cualquier/cosa.png"
    />
    <GifGridItem
      id="edfr"
      key="edfr"
      title="cualquier cosa"
      url="http://localhost/cualquier/cosa.png"
    />
  </section>
</section>
`;

exports[`Pruebas en <GifGrid /> Prueba de carga del componente  1`] = `
<section
  className="gif"
>
  <section
    className="gif-title animate__animated animate__lightSpeedInRight"
  >
    <h4>
      Samurai X
    </h4>
  </section>
  <p
    className="animate__animated animate__flash"
  >
    Cargando data...
  </p>
  <section
    className="gif-grid"
  />
</section>
`;
